decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/utils/BarcodeAndBasesMask.java

```diff
@@ -0,0 +1,178 @@
+            basesMaskBuilder.setIndexOneIgnoreLength(barcodeBasesMask.getIndexOneIgnoreLength());
+        } else if (runBasesMask.getIndexOneIncludeLength() != null && barcodeBasesMask.getIndexOneIncludeLength() == null) {
+            throw new DataMismatchException("Unexpected state, runBasesMask = [" + runBasesMask.toString() + "],"
+                    + " barcodeBasesMask = [" + barcodeBasesMask.toString() + "]");
+        } else if (runBasesMask.getIndexOneIncludeLength() != null && barcodeBasesMask.getIndexOneIncludeLength() != null) {
+            basesMaskBuilder.setIndexOneIncludeLength(barcodeBasesMask.getIndexOneIncludeLength());
+            basesMaskBuilder.setIndexOneIgnoreLength(barcodeBasesMask.getIndexOneIgnoreLength());
+        } else {
+            throw new DataMismatchException("Unexpected state, runBasesMask = [" + runBasesMask.toString() + "],"
+                    + " barcodeBasesMask = [" + barcodeBasesMask.toString() + "]");
+        }
+
+        //index two
+        if (runBasesMask.getIndexTwoIncludeLength() == null && barcodeBasesMask.getIndexTwoIncludeLength() == null) {
+            basesMaskBuilder.setIndexTwoIncludeLength(null);
+            basesMaskBuilder.setIndexTwoIgnoreLength(null);
+        } else if (runBasesMask.getIndexTwoIncludeLength() == null && barcodeBasesMask.getIndexTwoIncludeLength() != null) {
+            basesMaskBuilder.setIndexTwoIncludeLength(null);
+            basesMaskBuilder.setIndexTwoIgnoreLength(null);
+        } else if (runBasesMask.getIndexTwoIncludeLength() != null && barcodeBasesMask.getIndexTwoIncludeLength() == null) {
+            basesMaskBuilder.setIndexTwoIncludeLength(null);
+            if (runBasesMask.getIndexOneIncludeLength() == null) {
+                basesMaskBuilder.setIndexTwoIgnoreLength(null);
+            } else {
+                basesMaskBuilder.setIndexTwoIgnoreLength(Integer.MAX_VALUE);
+            }
+        } else if (runBasesMask.getIndexTwoIncludeLength() != null && barcodeBasesMask.getIndexTwoIncludeLength() != null) {
+            basesMaskBuilder.setIndexTwoIncludeLength(barcodeBasesMask.getIndexTwoIncludeLength());
+            basesMaskBuilder.setIndexTwoIgnoreLength(barcodeBasesMask.getIndexTwoIgnoreLength());
+        } else {
+            throw new DataMismatchException("Unexpected state, runBasesMask = [" + runBasesMask.toString() + "],"
+                    + " barcodeBasesMask = [" + barcodeBasesMask.toString() + "]");
+        }
+
+        //read two
+        basesMaskBuilder.setReadTwoIncludeLength(runBasesMask.getReadTwoIncludeLength());
+        basesMaskBuilder.setReadTwoIgnoreLength(runBasesMask.getReadTwoIgnoreLength());
+
+        return basesMaskBuilder.createBasesMask();
+    }
+
+    public static BasesMask calculateBasesMask(Collection<Barcode> barcodes) throws DataMismatchException {
+        Set<BasesMask> bs = barcodes.stream().map(b -> calculateBasesMask(b)).collect(Collectors.toSet());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)